--- a/gfx/drivers_context/drm_go2_ctx.c
+++ b/gfx/drivers_context/drm_go2_ctx.c
@@ -295,6 +295,7 @@
 
 static void gfx_ctx_go2_drm_swap_buffers(void *data)
 {
+   settings_t *settings                 = config_get_ptr();
    gfx_ctx_go2_drm_data_t 
       *drm   = (gfx_ctx_go2_drm_data_t*)data;
 
@@ -321,11 +322,21 @@
    go2_context_swap_buffers(drm->context);
 
    go2_surface_t* surface = go2_context_surface_lock(drm->context);
+
+if (settings->bools.video_oga_vertical_enable) {
+   go2_presenter_post(drm->presenter,
+         surface,
+         src_x, src_y, src_w, src_h,
+         out_y, out_x, out_h, out_w,
+         GO2_ROTATION_DEGREES_0, 2);
+} else {
    go2_presenter_post(drm->presenter,
          surface,
          src_x, src_y, src_w, src_h,
          out_y, out_x, out_h, out_w,
          GO2_ROTATION_DEGREES_270, 2);
+}
+
    go2_context_surface_unlock(drm->context, surface);
 #endif
 }
